### PR TITLE
Support for posting location to Twitter and Facebook

### PR DESCRIPTION
**Pull request by MariusVolkhart**
@jault3 @atscott please review

### CODE REVIEW

**Comment by jault3** [comment]
@MariusVolkhart @atscott after talking with Marius a few days ago, we decided to do this properly, we would need more support for regular users. This has changed a bit. The tokens no longer need to be sent because when you login or link an account, the API will remember them for you. The postToTw or postToFb tells whether or not to post to the respective social media site if you put any of your trucks into serving mode. This would also require some database changes, which I've already mapped out. Unfortunately it is hard to show those here because it doesn't show in the protos. 

Let's discuss this and see if we like it and our direction forward. Perhaps we should meet tomorrow at our regular time and iron this out.

**Comment by atscott** [comment]
So the AuthRequest now returns a User as a response. What authorization token are we supposed to use for following requests when logged in?

**Comment by jault3** [comment]
@atscott you are correct. I also completely ignored backwards compatibility. Now that I fixed it, the API is going to have to support all of the fields in there for awhile.

File: com/truckmuncher/api/user.proto [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+service UserService {
+
+    /**
+     * Link another social media account to your existing account. This can be used by regular
+     * users or vendors.
+     *
+     * This call requires authorization
+     */
+    rpc linkAccount (LinkAccountRequest) returns (User);
```
Comment (MariusVolkhart) [inline com/truckmuncher/api/user.proto]: wasn't sure how this was going to work, but the API code clears it up. almost creepy how we can do that without it seeming like we need to send data :P

**Comment by atscott** [comment]
Looks good to me

**Comment by MariusVolkhart** [comment]
:shipit:

**Pull request merged**